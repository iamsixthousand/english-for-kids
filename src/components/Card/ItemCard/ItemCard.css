.itemCard.front {
    height: 100%;
    border-radius: 10px;
    opacity: 0.9;
    overflow: hidden;
    
}
.itemCard.front:hover {
    box-shadow: 0 8px 32px 0 rgba(51, 48, 34, 0.4);
    transition: box-shadow 0.2s ease-in-out;
    opacity: 1;
}

.itemCard.back {
    background-color: aquamarine;
    position: absolute;
    
}


.CardImageContainer {
    height: 70%;
    display: flex;
    justify-content: center;
    align-items: center;
}

.CardImage {
    width: 100%;
    height: 100%;
}

.CardControlPanel {
    height: 30%;
    background-color: grey;
    display: grid;
    justify-content: center;
    align-items: center;
    text-transform: uppercase;
}