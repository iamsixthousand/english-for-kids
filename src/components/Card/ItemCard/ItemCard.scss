@import "../../../index.scss";
@import "../../../styles.scss";

.CardContainer {
  width: 100%;
  height: 250px;
  background-color: white;
  align-self: center;
  justify-self: center;
  border-radius: 10px;
  position: relative;
  transition: transform 0.5s ease-in-out;
  transform-style: preserve-3d;

  &.flipped {
    transform: rotateY(180deg);
  }
}

.ItemCard {
  position: absolute;
  height: 100%;
  width: 100%;
  border-radius: 10px;
  opacity: 0.9;
  overflow: hidden;
  transition: all 0.15s ease-in-out;
  box-shadow: $box-shadow-small;
  backface-visibility: hidden;

  &:hover {
    box-shadow: $box-shadow-big;
    transform: scale(1.005);
    opacity: 1;
  }

  &.back {
    transform: rotateY(180deg);
  }
}

.CardImageContainer {
  height: 80%;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: height 0.3s ease-in-out;

  &.play {
    height: 100%;
  }
}

.CardImage {
  width: 100%;
  height: 100%;
  object-fit: cover;
  pointer-events: none;
  user-select: none;
}

.CardControlPanel {
  height: 20%;
  background-color: $control-panelbg;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  align-items: center;
  text-transform: uppercase;

  &.flipped {
    grid-template-columns: repeat(1, 1fr);
  }
}

.ControlPanelPlayButton {
  height: 35px;
  width: 35px;
  align-self: center;
  justify-self: center;
  cursor: pointer;
  user-select: none;

  &:hover {
    animation: swing 0.6s ease;
    animation-iteration-count: 1;
  }
}

.ControlPanelEyeButton {
  height: 50px;
  width: 50px;
  align-self: center;
  justify-self: center;
  cursor: pointer;
  user-select: none;

  &:hover {
    animation: swing 0.6s ease;
    animation-iteration-count: 1;
  }
}

.ControlPanelText {
  font-size: 25px;
  font-weight: $bold;
  user-select: none;
}
