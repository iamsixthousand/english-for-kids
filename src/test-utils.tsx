import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import gameProcessReducer from './state/gameProcessReducer';

const WithStoreProvider: React.FC = ({ children }) => {
  const myStore = createStore(
    combineReducers({
      gameProcess: gameProcessReducer,
    })
  );
  return <Provider store={myStore}>{children}</Provider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: WithStoreProvider, ...options });

export * from '@testing-library/react';
export { customRender as render };
