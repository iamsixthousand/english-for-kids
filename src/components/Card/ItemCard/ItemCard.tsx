import React from 'react';
import { ItemCardProps } from '../../../interfaces/interfaces';
import './ItemCard.css';

export const ItemCard: React.FC<ItemCardProps> = ({
  isPlaying,
  word,
  translation,
  audio,
  image,
}) => {
  return (
    <div className="itemCard">
      {isPlaying}
      <div>{image}</div>
      <div>{word}</div>
      <div>{translation}</div>
      <div>{audio}</div>
    </div>
  );
};
