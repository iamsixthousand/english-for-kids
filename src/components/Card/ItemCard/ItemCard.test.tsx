/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import React from 'react';
import { screen } from '@testing-library/react';
import { ItemCard } from './ItemCard';
import { cards } from '../../../cardData';
import { renderWithStore } from '../../../test-utils';

describe('ItemCard component:', () => {
  const mockGameStepsFunc = jest.fn();
  const { word } = cards[0][0]; // cry
  const { image } = cards[0][0];
  const { translation } = cards[0][0]; // плакать
  const audio = cards[0][0].audioSrc;

  const renderComp = (
    <ItemCard
      word={word}
      translation={translation}
      audio={audio}
      image={image}
      inGameAnswer={mockGameStepsFunc}
    />
  );

  it('should contain correct word text', () => {
    renderWithStore(renderComp);
    expect(screen.getByText('cry')).toBeInTheDocument();
  });
  it('should contain correct translation text', () => {
    renderWithStore(renderComp);
    expect(screen.getByText('плакать')).toBeInTheDocument();
  });
  it('should contain 4 image elements', () => {
    renderWithStore(renderComp);
    expect(screen.getAllByRole('img')).toHaveLength(4);
  });
  it('should contain 4 button elements', () => {
    renderWithStore(renderComp);
    expect(screen.getAllByRole('button')).toHaveLength(4);
  });
  it('matches snapshot', () => {
    const categoryCard = renderWithStore(renderComp);
    expect(categoryCard).toMatchSnapshot();
  });
});
