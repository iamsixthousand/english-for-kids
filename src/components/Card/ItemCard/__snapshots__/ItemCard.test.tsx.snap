// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard component: matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-role="Mask"
      >
        <div
          class="CardContainer"
        >
          <div
            class="ItemCard front"
          >
            <div
              class="CardImageContainer"
              data-word="cry"
              role="button"
              tabindex="0"
            >
              <img
                alt="cry"
                class="CardImage"
                src="/img/cry.jpg"
              />
            </div>
            <div
              class="CardControlPanel"
            >
              <button
                style="background-color: rgb(248, 241, 241);"
                type="button"
              >
                <img
                  alt="play-sound"
                  class="ControlPanelPlayButton"
                  src="/play-button.png"
                />
              </button>
              <span
                class="ControlPanelText"
              >
                cry
              </span>
              <button
                style="background-color: rgb(248, 241, 241);"
                type="button"
              >
                <img
                  alt="watch-translation"
                  class="ControlPanelEyeButton"
                  src="/eye-button.png"
                />
              </button>
            </div>
          </div>
          <div
            class="ItemCard back"
          >
            <div
              class="CardImageContainer"
              role="button"
              tabindex="0"
            >
              <img
                alt="cry"
                class="CardImage"
                src="/img/cry.jpg"
              />
            </div>
            <div
              class="CardControlPanel flipped"
            >
              <span
                class="ControlPanelText"
              >
                плакать
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-role="Mask"
    >
      <div
        class="CardContainer"
      >
        <div
          class="ItemCard front"
        >
          <div
            class="CardImageContainer"
            data-word="cry"
            role="button"
            tabindex="0"
          >
            <img
              alt="cry"
              class="CardImage"
              src="/img/cry.jpg"
            />
          </div>
          <div
            class="CardControlPanel"
          >
            <button
              style="background-color: rgb(248, 241, 241);"
              type="button"
            >
              <img
                alt="play-sound"
                class="ControlPanelPlayButton"
                src="/play-button.png"
              />
            </button>
            <span
              class="ControlPanelText"
            >
              cry
            </span>
            <button
              style="background-color: rgb(248, 241, 241);"
              type="button"
            >
              <img
                alt="watch-translation"
                class="ControlPanelEyeButton"
                src="/eye-button.png"
              />
            </button>
          </div>
        </div>
        <div
          class="ItemCard back"
        >
          <div
            class="CardImageContainer"
            role="button"
            tabindex="0"
          >
            <img
              alt="cry"
              class="CardImage"
              src="/img/cry.jpg"
            />
          </div>
          <div
            class="CardControlPanel flipped"
          >
            <span
              class="ControlPanelText"
            >
              плакать
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
